libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshController.kt

```diff
@@ -26,29 +32,63 @@ import kotlin.coroutines.resume
  * This class is responsible for refreshing the current direction route's traffic.
  * This does not support alternative routes.
  */
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
 internal class RouteRefreshController(
     private val routeRefreshOptions: RouteRefreshOptions,
     private val routeRefresh: RouteRefresh,
     private val currentIndicesProvider: CurrentIndicesProvider,
     private val routeDiffProvider: DirectionsRouteDiffProvider = DirectionsRouteDiffProvider(),
-    private val localDateProvider: () -> Date
+    threadController: ThreadController,
+    private val localDateProvider: () -> Date,
 ) {
 
+    private val mainJobControl = threadController.getMainScopeAndRootJob()
+
+    @RouteRefreshExtra.RouteRefreshState
+    private var state: String = RouteRefreshExtra.REFRESH_STATE_IDLE
+        set(value) {
+            if (field == value) return
+            field = value
+            val _observer = observer ?: return
+            mainJobControl.scope.launch { _observer.onNewState(value) }
```

Comment:
why is this required? can't you just trigger observer?

Reply:
actually, I did it as we often spread data via observer via Main thread (like `RoutesObserver` and some other observers)

Reply:
route refresh controller works only with UI thread, doesn't it?

Reply:
actually it is, but it's not guaranty (the only `suspend` keyword) and make it explitly seems reasonable